cmd/harbor/root/registry/update.go

```diff
@@ -7,82 +7,67 @@ import (
 	"github.com/goharbor/go-client/pkg/sdk/v2.0/client/registry"
 	"github.com/goharbor/go-client/pkg/sdk/v2.0/models"
 	"github.com/goharbor/harbor-cli/pkg/utils"
-	"github.com/goharbor/harbor-cli/pkg/views/registry/create"
+	"github.com/goharbor/harbor-cli/pkg/views/registry/update"
 	log "github.com/sirupsen/logrus"
 	"github.com/spf13/cobra"
 	"github.com/spf13/viper"
 )
 
 // NewUpdateRegistryCommand creates a new `harbor update registry` command
 func UpdateRegistryCommand() *cobra.Command {
-	var opts create.CreateView
 
+	var opts models.Registry
 	cmd := &cobra.Command{
 		Use:   "update",
 		Short: "update registry",
 		Args:  cobra.MaximumNArgs(1),
 		Run: func(cmd *cobra.Command, args []string) {
 			var err error
 			var registryId int64
+			credentialName := viper.GetString("current-credential-name")
+			client := utils.GetClientByCredentialName(credentialName)
+			ctx := context.Background()
+			if len(args) > 0 {
+				registryId, err = strconv.ParseInt(args[0], 10, 64)
+			} else {
+				registryId = utils.GetRegistryNameFromUser()
+			}
+			if err != nil {
+				log.Errorf("failed to parse registry id: %v", err)
+			}
 
-			updateView := &create.CreateView{
+			response, err := client.Registry.GetRegistry(ctx,&registry.GetRegistryParams{ID: registryId})
+			if err != nil {
+				log.Fatal(err)
+			}
+			opts = *response.GetPayload()
+			
+			
+			updateView := &models.Registry{
 				Name:        opts.Name,
 				Type:        opts.Type,
 				Description: opts.Description,
 				URL:         opts.URL,
-				Credential: create.RegistryCredential{
+				Credential: &models.RegistryCredential{
 					AccessKey:    opts.Credential.AccessKey,
 					Type:         opts.Credential.Type,
 					AccessSecret: opts.Credential.AccessSecret,
 				},
 				Insecure: opts.Insecure,
 			}
 
-			if len(args) > 0 {
-				registryId, err = strconv.ParseInt(args[0], 10, 64)
-			} else {
-				registryId = utils.GetRegistryNameFromUser()
-			}
-
-			if err != nil {
-				log.Errorf("failed to parse registry id: %v", err)
-			}
-
-			if opts.Name != "" && opts.Type != "" && opts.URL != "" {
-				err = runUpdateRegistry(updateView, registryId)
-			} else {
-				err = updateRegistryView(updateView, registryId)
-			}
-
+			update.UpdateRegistryView(updateView)
+			err = runUpdateRegistry(updateView, registryId)
 			if err != nil {
 				log.Errorf("failed to update registry: %v", err)
 			}
 		},
 	}
 
-	flags := cmd.Flags()
```

Comment:
Thanks for your contribution! Could you please add back the below flags? They're crucial for  automation and scripting processes, and their absence could potentially disrupt CI/CD pipelines.

Thanks.

Reply:
Yes. I will add those flags.